### ISSUE TITLE
Error occured at every run "Expected DataContext to already have been set via setCtx. " and the enw changes does not be saved on Cypress

### ISSUE THREAD

**Issue by Eyatester** [cypress-io/cypress#29637]
### Current behavior

When running my code this error appears : also when we add changes on the code they are not saved 
```
GET /chrome-variations/seed?osname=win&channel=stable&milestone=125 - - ms - -
Error:
      Expected DataContext to already have been set via setCtx. If this is a
      testing context, make sure you are calling "setCtx" in a before hook,
      otherwise check the application flow.
    at c (<embedded>:4419:89919)
    at Object.u (<embedded>:4649:291801)
    at Object.<anonymous> (<embedded>:4669:29094)
    at Object.tryCatcher (C:\Users\DELL\AppData\Local\Cypress\Cache\12.8.1\Cypress\resources\app\node_modules\bluebird\js\release\util.js:16:23)
    at Object.execute (C:\Users\DELL\AppData\Local\Cypress\Cache\12.8.1\Cypress\resources\app\node_modules\bluebird\js\release\method.js:15:34)
    at C.close (<embedded>:4721:423879)
```
### Desired behavior

how to fix this error and changes should be saved

### Test code to reproduce
```
GET /chrome-variations/seed?osname=win&channel=stable&milestone=125 - - ms - -
Error:
      Expected DataContext to already have been set via setCtx. If this is a
      testing context, make sure you are calling "setCtx" in a before hook,
      otherwise check the application flow.
    at c (<embedded>:4419:89919)
    at Object.u (<embedded>:4649:291801)
    at Object.<anonymous> (<embedded>:4669:29094)
    at Object.tryCatcher (C:\Users\DELL\AppData\Local\Cypress\Cache\12.8.1\Cypress\resources\app\node_modules\bluebird\js\release\util.js:16:23)
    at Object.execute (C:\Users\DELL\AppData\Local\Cypress\Cache\12.8.1\Cypress\resources\app\node_modules\bluebird\js\release\method.js:15:34)
    at C.close (<embedded>:4721:423879)
```
### Cypress Version

12.8.1

### Node version

20.13.1

### Operating System

windows  64

**Comment by jennifer-shehane** [comment]
@Eyatester Could you update to the latest version of Cypress and let us know if this error still occurs. This issue may already be fixed.

**Comment by BreakBB** [comment]
I just started to receive the same error @jennifer-shehane 

The error only appeared after adding debug logging: `DEBUG=cypress:server:util:process_profiler cypress run --browser chrome` AND it only appears when running in a docker container. Locally I do not see the error and I do see the debug logging, while in the container I do see the error, but no logging.

I am using Cypress `v13.13.0` and as Docker image I tested both `cypress/browsers:node-20.12.2-chrome-124.0.6367.60-1-ff-125.0.1-edge-124.0.2478.51-1` and also `cypress/browsers:node-22.0.0-chrome-124.0.6367.60-1-ff-125.0.2-edge-124.0.2478.51-1`.
